pgjdbc/src/test/java/org/postgresql/test/jdbc3/StringTypeParameterTest.java

```diff
@@ -17,15 +22,11 @@
 import java.sql.Types;
 import java.util.Properties;
 
-public class StringTypeParameterTest extends TestCase {
+public class StringTypeParameterTest {
 
   private Connection _conn;
 
-  public StringTypeParameterTest(String name) {
-    super(name);
-  }
-
-  protected boolean setUp(String stringType) throws Exception {
+  private boolean prepare(String stringType) throws Exception {
```

Comment:
The replacement of `setUp` with `prepare` does break things.

For instance, `testParameterStringTypeVarchar` has the following code:
```
  @Test
  public void testParameterStringTypeVarchar() throws Exception {
    if (!TestUtil.isProtocolVersion(_conn, 3)) {
      return;
    }
    testParameterVarchar("varchar");
  }
```

The thing is `TestUtil.isProtocolVersion` check should be performed when connection is active. When `setUp` got replaced with `prepare`, the connection became `null`, and `isProtocolVersion` became always false.